contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_utils.py

```diff
@@ -0,0 +1,157 @@
+import logging
+import os
+import shlex
+import subprocess
+import uuid
+
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from checkbox_support.scripts.psnr import get_average_psnr
+
+GST_LAUNCH_BIN = os.getenv("GST_LAUNCH_BIN", "gst-launch-1.0")
+PLAINBOX_SESSION_SHARE = os.getenv("PLAINBOX_SESSION_SHARE", "/var/tmp")
+VIDEO_CODEC_TESTING_DATA = os.getenv("VIDEO_CODEC_TESTING_DATA")
+if not VIDEO_CODEC_TESTING_DATA or not os.path.exists(
+    VIDEO_CODEC_TESTING_DATA
+):
+    raise SystemExit(
+        "Error: Please define the proper path of golden sample folder to "
+        "the environment variable 'VIDEO_CODEC_TESTING_DATA'"
+    )
+# Folder stores the golden samples
+SAMPLE_2_FOLDER = "sample_2_big_bug_bunny"
+
+
+class GStreamerEncodePlugins(Enum):
+    V4L2H264ENC = "v4l2h264enc"
+    V4L2H265ENC = "v4l2h265enc"
+    V4L2JPEGENC = "v4l2jpegenc"
+
+
+class GStreamerMuxerType(Enum):
+    """
+    An enumeration representing different GStreamer muxers and their
+    corresponding file extensions.
+
+    Attributes:
+        MP4MUX (str): Represents the MP4 muxer, associated with the 'mp4' file
+                        extension.
+        AVIMUX (str): Represents the AVI muxer, associated with the 'avi' file
+                        extension.
+        MATROSKAMUX (str): Represents the Matroska muxer, associated with the
+                        'mkv' file extension.
+    """
+
+    MP4MUX = "mp4"
+    AVIMUX = "avi"
+    MATROSKAMUX = "mkv"
+
+    @classmethod
+    def get_extension(cls, mux_type):
+        if mux_type.upper() in cls.__members__:
+            return cls[mux_type].value
+        else:
+            raise ValueError(
+                "Invalid mux type: {}."
+                " Available types are: {}".format(
+                    mux_type, ", ".join(cls.__members__.keys())
+                )
+            )
+
+
+def execute_command(cmd: str = "", timeout: int = 300) -> str:
+    """
+    Executes the GStreamer command and extracts the specific data from the
+    output. The specific data is the value of last-message which is exposed by
+    fpsdisplaysink.
+
+    :param cmd:
+        The GStreamer command to execute.
+
+    :returns:
+        The extracted last_message.
+    """
+    try:
+        logging.info("Starting command: '{}'".format(cmd))
+        ret = subprocess.run(
+            shlex.split(cmd),
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            encoding="utf-8",
```

Comment:
Property `encoding` doesn't exist in python 3.5 (not an issue if we are guaranteed python > 3.6

Reply:
Good catch. Thanks